### PR TITLE
GraphQL-272: [My Account] Added test address extractor

### PR DESCRIPTION
**Pull request by AleksLi**
### Description (*)

Created a new PR for Create new address through GraphQL request. Started from Validate address on Created Address action.

### Fixed Issues (if relevant)

1. magento/magento2#<[272](https://github.com/magento/graphql-ce/issues/272)>: [My Account] Add support of Customer Address attributes

### GraphQL test query:
```
createCustomerAddress (
    input: {
      region:{
        region:"Arizona",
        region_id:4,
        region_code:"AZ"
      },
      firstname:"Oli",
      lastname: "Ne",
      telephone:"123456789",
      street: "123 as a look ek",
      city:"CPH",
      country_id:DK,
      postcode:"1600",
      default_shipping: true,
      default_billing:true,
      custom_attributes: {
        attribute_code: "de",
        value:"new dan"
      }
    }
  ) {
    id
    customer_id
  }
```

Please give me your opinion and tell me if I'm doing something wrong.

### CODE REVIEW

**Comment by naydav** [comment]
Hello @AleksLi,
Please check failed travis build
Also, need to cover functionality with API-functional tests
Thanks

**Comment by AleksLi** [comment]
@naydav  Am I get the idea correctly?
Thank you. I wasn't sure I did everything right, that's why I didn't complete everything here. Now I will proceed and make sure the builds are green and api-functional tests are green

**Comment by AleksLi** [comment]
@naydav I don't get what is wrong with the Dashboard:tunnel method in the builds. I think that's the common issue. I saw that in my other PRs and they disappeared suddenly somehow. Can you take a look, please?

**Comment by naydav** [comment]
Hi @AleksLi,
Will you continue to work on the ticket?  Now it looks like the created code is not used anywhere

File: app/code/Magento/CustomerGraphQl/Model/Customer/Address/CustomerAddressCreateDataValidator.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+     * @throws \Exception
+     */
+    public function validate(array $addressData): void
+    {
+        $errors = $this->customerAddressValidator->validateAddress($addressData);
+
+        $errorInput = [];
+
+        if ($errors !== true) {
+            foreach ($errors as $messageText) {
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Customer/Address/CustomerAddressCreateDataValidator.php:44]: What is the difference between `$errors` and `$errorInput`?

File: app/code/Magento/CustomerGraphQl/Model/Customer/Address/Validator.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+     * @param array $addressData
+     * @return array|bool
+     */
+    public function validateAddress($addressData)
+    {
+        $request = $this->createNewRequest();
+        $request->setParams($addressData);
+
+        /** @var Form $addressForm */
+        $addressForm = $this->formFactory->create(
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Customer/Address/Validator.php:49]: Could you check how it's work in REST api... looks not good when we use something like `form` in HTTP calls responsible area

File: app/code/Magento/CustomerGraphQl/Model/Customer/Address/Validator.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        return $messages ?? true;
+    }
+
+    /**
+     * @return RequestInterface
+     */
+    private function createNewRequest()
+    {
+        return $this->requestFactory->create();
```
Comment (naydav) [inline app/code/Magento/CustomerGraphQl/Model/Customer/Address/Validator.php:80]: Redundant method

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/UpdateCustomerAddressTest.php [hunk 4]
```diff
@@ -187,6 +187,9 @@ public function testUpdateCustomerAddressWithMissingAttribute()
   }
 }
 MUTATION;
+
```
Comment (naydav) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Customer/UpdateCustomerAddressTest.php:190]: Looks like it'is not enough for testing ))

**Review by naydav — CHANGES_REQUESTED** [review]

**Comment by naydav** [comment]
Hello @AleksLi,
PR was closed due to inactivity. Please reopen and update if you wish to continue. 
Thanks!

**Pull request closed without merging**